assayist/client/modify.py

```diff
@@ -0,0 +1,103 @@
+# SPDX-License-Identifier: GPL-3.0+
+
+import neomodel
+
+from assayist.common.models.source import Component
+
+
+@neomodel.db.transaction
+def set_component_names(c_name, c_type, c_namespace='', alternatives=None):
+    """
+    Create, update, and/or merge components to ensure one canonical component remains.
+
+    :param str c_name: the canonical name of the component
+    :param str c_type: the canonical type of the component
+    :param str c_namespace: the canonical namespace of the component
+    :raises ValueError: if the input isn't the proper type or c_name's value is also in alternatives
+    """
+    for arg in (c_name, c_type, c_namespace):
+        if not isinstance(arg, str):
+            raise ValueError('c_name, c_type, and c_namespace must be strings')
+
+    if alternatives is None:
+        alternatives = []
+    if not isinstance(alternatives, list) and not isinstance(alternatives, tuple):
+        raise ValueError('The alternatives keyword argument must be None, a list, or a tuple')
+
+    for alternative in alternatives:
+        if not isinstance(alternative, str):
+            raise ValueError('The alternatives keyword argument must only contain strings')
+
+    if c_name in alternatives:
+        raise ValueError('The canonical name can\'t also be an alternative')
+
+    # Create a new list with the canonical name and alternatives
+    names = [c_name] + alternatives
```

Comment:
If alternatives is passed as a tuple, this will fail. Better to use `itertools.chain` to concatenate two iterables.